components/brave_new_tab_ui/reducers/new_tab_reducer.ts

```diff
@@ -384,6 +385,201 @@ export const newTabReducer: Reducer<NewTab.State | undefined> = (state: NewTab.S
       }
       break
 
+    case types.SET_HIDE_BALANCE:
+      state = { ...state }
+      state.binanceState.hideBalance = payload.hide
+      break
+
+    case types.ON_BINANCE_ACCOUNT_BALANCES:
+      const { balances } = payload
+      if (!balances) {
+        break
+      }
+
+      state = { ...state }
+      state.binanceState.accountBalances = balances
+
+      let totalBTC = 0.00
+      let totalBTCUSDValue = 0.00
+
+      const btcPrice = state.binanceState.btcPrice
+      const btcValues = state.binanceState.assetBTCValues
+
+      Object.keys(balances).map((symbol: string) => {
+        const balance = balances[symbol]
+        if (symbol === 'BTC') {
+          totalBTC += parseFloat(balance)
+          totalBTCUSDValue += parseFloat(getUSDPrice(balance, btcPrice))
+        } else {
+          const totalInBTC = parseFloat(btcValues[symbol]) * parseFloat(balance)
+          const totalAssetValue = getUSDPrice(totalInBTC.toString(), btcPrice)
+          totalBTC += totalInBTC
+          totalBTCUSDValue += parseFloat(totalAssetValue)
+        }
+      })
+
+      state.binanceState = {
+        ...state.binanceState,
+        accountBTCValue: totalBTC.toString(),
+        accountBTCUSDValue: totalBTCUSDValue.toFixed(2).toString()
+      }
+
+      break
+
+    case types.ON_VALID_AUTH_CODE:
+      state = { ...state }
+      state.binanceState.userAuthed = true
+      state.binanceState.authInProgress = false
+      break
+
+    case types.DISCONNECT_BINANCE:
+      state = { ...state }
+      state.binanceState = {
+        ...storage.defaultState.binanceState,
+        binanceSupported: true
+      }
+      break
+
+    case types.CONNECT_TO_BINANCE:
+      state = { ...state }
+      state.binanceState.authInProgress = true
+      break
+
+    case types.ON_BINANCE_CLIENT_URL:
+      state = { ...state }
+      state.binanceState.binanceClientUrl = payload.clientUrl
```

Comment:
maybe add some validation that this URL is HTTPS and connects to an expected hostname?